### ISSUE TITLE
TypeError: undefined is not an object (evaluating 'env().platform.fetch') in createDetector using tfjs-react-native with pose-detection

### ISSUE THREAD

**Issue by darringrove** [tensorflow/tfjs#6611]
**System information**
Have I written custom code (as opposed to using a stock [example](https://github.com/tensorflow/tfjs-examples/tree/master/react-native/pose-detection) script provided in TensorFlow.js):
No. This issue is reproduced with a bare RN app + the tfjs RN pose detection example. I fixed some typescript issues in the example but did not fundamentally change it.

OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
macOS Monterey 12.4, Xcode 13.4.1

Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
iPhone 12 Pro, iOS 15.5

TensorFlow.js installed from (npm or script link): yarn
TensorFlow.js version (use command below): 3.18.0
I was careful to use the same versions of all packages as the example
```
    "@mediapipe/pose": "~0.4.0",
    "@react-native-async-storage/async-storage": "~1.17.3",
    "@tensorflow-models/pose-detection": "2.0.0",
    "@tensorflow/tfjs": "3.18.0",
    "@tensorflow/tfjs-react-native": "0.8.0",
    "expo": "~45.0.6",
    "expo-camera": "^12.2.0",
    "expo-file-system": "^14.0.0",
    "expo-gl": "^11.3.0",
    "expo-gl-cpp": "^11.3.0",
    "expo-screen-orientation": "~4.2.0",
    "react": "17.0.2",
    "react-dom": "17.0.2",
    "react-native": "0.68.2",
    "react-native-fs": "2.14.1",
    "react-native-gesture-handler": "~2.2.1",
    "react-native-svg": "12.3.0"
```
Browser version: n/a
Tensorflow.js Converter Version: n/a

**Describe the current behavior**
on the call to createDetector
```
WARN Possible Unhandled Promise Rejection (id: 0):
TypeError: undefined is not an object (evaluating 'env().platform.fetch')
```

Note: call `await tf.ready()` is successful

**Describe the expected behavior**
createDetector should return the model

**Standalone code to reproduce the issue Provide a reproducible test case that is the bare minimum necessary to generate the problem.**
[darringrove/posetest](https://github.com/darringrove/posetest)
```
git clone repo
cd posetest
yarn
npx pod-install
```
Run in Xcode on iPhone

**Other info / logs Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.**

I have verified that the [example](https://github.com/tensorflow/tfjs-examples/tree/master/react-native/pose-detection) using RN with pose detection works using Expo Go on my machine.

I believe this is an issue related to starting with a bare RN app.

1. I created a new RN project using the React Native CLI and the typescript template for RN 0.68.2.
2. I followed the [setup instructions](https://github.com/tensorflow/tfjs/tree/master/tfjs-react-native) in tfjs-react-native. There is a key difference to note (found by following dependent instructions):
 - react-native-unimodules is deprecated and replace by expo. I used the automatic installation for expo.
3. I added additional packages used by the RN pose detection example.
4. I copied the code from App.tsx from the example (and fixed some Typescript issues for a clean build).

The issue is recreated in [darringrove/posetest](https://github.com/darringrove/posetest)

Thanks to @jinjingforever for updating the packages in the examples recently. I'd be happy to add this React Native CLI example to the examples project once this issue is resolved.

Thanks in advance!

**Comment by darringrove** [comment]
@rthadur If it's helpful, the error is being thrown in HTTPRequest in tfjs-core. In the case of the example, it's being called with:
 path = https://tfhub.dev/google/tfjs-model/movenet/singlepose/lightning/4/model.json?tfjs-format=file
 loadOptions = {fromTFHub: true}

It gets to `this.fetch = env().platform.fetch;` and env().platform is undefined.

**Comment by darringrove** [comment]
@rthadur @jinjingforever Since this issue involved loading the model, I decided to try the LOAD_MODEL_FROM_BUNDLE option. In this case, createDetector successfully returns the new model, but the call to estimatePoses is taking forever to return.

**Comment by darringrove** [comment]
@rthadur Any update on this? Were you able to reproduce it?

**Comment by rthadur** [comment]
Sorry for the delay , let's wait to hear from Jing .

**Comment by Wil2129** [comment]
Hi @darringrove, any update on this?

I am facing the same issue, using a bare RN app (with expo instread of the deprecated react-native-unimodules). Did you manage to make it work? If yes, how?

**Comment by Wil2129** [comment]
I think I might have a work around:

**TL;DR:** Add the line `import '@tensorflow/tfjs-react-native/dist/platform_react_native';` in the file where you load your model

**Explanation:**
I did a little digging to understand what was going on. Normally tfjs defines two platforms, node and browser. Tfjs RN defines its own platform (RN). That's why the `import '@tensorflow/tfjs-react-native';` is import. The issue comes from this import. In its index.js we have

> import './platform_react_native';
export { asyncStorageIO } from './async_storage_io';
export { bundleResourceIO } from './bundle_resource_io';
export { decodeJpeg } from './decode_image';
export { fetch } from './platform_react_native';
export { version } from './version';
export * from './camera/camera';

The important line here is `import './platform_react_native';`. This is the file where the RN platform is actually defined.
> tf.env().registerFlag('IS_REACT_NATIVE', () => navigator && navigator.product === 'ReactNative');
if (tf.env().getBool('IS_REACT_NATIVE')) {
    setupGlobals();
    registerWebGLBackend();
    tf.setPlatform('react-native', new PlatformReactNative());
}

But now when importing _tensorflow/tfjs-react-native_, it seems like it doesn't subsequently import the _platform_react_native.js_ file. So if you do `import '@tensorflow/tfjs-react-native/dist/platform_react_native';`, it actually execute the part of the code where the RN platform is defined and thus we get a non-null plaftorm and _platform.fetch_.

I'm not sure if this has to do with this package directly, I think it revolves around how packages are imported/executed by RN.

**Comment by darringrove** [comment]
@Wil2129 Thanks for the research and work around. To answer your original question, this issue has not been resolved and is still open. I temporarily recast my project as a managed Expo app to move forward, but I'm hoping for a resolution so that I can return to a bare RN app.

**Comment by gaikwadrahul8** [comment]
Hi, @darringrove 

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The TFJs team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TFJs [version](https://github.com/tensorflow/tfjs/releases/tag/tfjs-v4.10.0) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tfjs/blob/master/README.md) to stay up to date with the latest developments which are happening in the Tensorflow.js space.

Thank you for your support and cooperation.
